Modules/_threadmodule.c

```diff
@@ -1184,13 +1184,17 @@ thread_PyThread_start_new_thread(PyObject *self, PyObject *fargs)
         return NULL;
     }
 
-    struct bootstate *boot = PyMem_NEW(struct bootstate, 1);
+    // gh-109795: Use PyMem_RawMalloc() to allocate new thread bootstate.
+    // If new thread is started at Python finalization,
+    // i.e. thread_run() routine is invoked in newly created OS thread,
+    // it should be able to free bootstate without holding the GIL.
```

Comment:
```suggestion
    // gh-109795: Use PyMem_RawMalloc() instead of PyMem_Malloc(),
    // because it should be possible to call thread_bootstate_free()
    // without holding the GIL.
```